src/util/secrets/secrets.c

```diff
@@ -937,6 +939,93 @@ static char *local_dn_to_path(TALLOC_CTX *mem_ctx,
     return path;
 }
 
+/* Complete list of ccache names(UUID:name) */
+errno_t sss_sec_list_cc_uuids(TALLOC_CTX *mem_ctx,
+                              struct sss_sec_ctx *sec,
+                              const char ***_uuid_list,
+                              const char ***_uid_list,
+                              size_t *_uuid_list_count)
```

Comment:
> Hi Justin. Thank you for the changes, the design looks good now. Its nice to see how the amount of code got reduced. There are lots of talloc issues that needs to be fixed, otherwise just few minor comments.

Thanks very much for the detailed review. I just pushed a new version which should address each review item aside from 1 or 2 items - for those I replied in the conversation for further clarification.